logtail/logtail.go

```diff
@@ -44,17 +46,18 @@ type Encoder interface {
 }
 
 type Config struct {
-	Collection     string           // collection name, a domain name
-	PrivateID      PrivateID        // machine-specific private identifier
-	BaseURL        string           // if empty defaults to "https://log.tailscale.io"
-	HTTPC          *http.Client     // if empty defaults to http.DefaultClient
-	SkipClientTime bool             // if true, client_time is not written to logs
-	LowMemory      bool             // if true, logtail minimizes memory use
-	TimeNow        func() time.Time // if set, subsitutes uses of time.Now
-	Stderr         io.Writer        // if set, logs are sent here instead of os.Stderr
-	StderrLevel    int              // max verbosity level to write to stderr; 0 means the non-verbose messages only
-	Buffer         Buffer           // temp storage, if nil a MemoryBuffer
-	NewZstdEncoder func() Encoder   // if set, used to compress logs for transmission
+	Collection        string           // collection name, a domain name
+	PrivateID         PrivateID        // machine-specific private identifier
+	BaseURL           string           // if empty defaults to "https://log.tailscale.io"
+	HTTPC             *http.Client     // if empty defaults to http.DefaultClient
+	SkipClientTime    bool             // if true, client_time is not written to logs
+	LowMemory         bool             // if true, logtail minimizes memory use
+	TimeNow           func() time.Time // if set, subsitutes uses of time.Now
+	Stderr            io.Writer        // if set, logs are sent here instead of os.Stderr
+	StderrLevel       int              // max verbosity level to write to stderr; 0 means the non-verbose messages only
+	Buffer            Buffer           // temp storage, if nil a MemoryBuffer
+	NewZstdEncoder    func() Encoder   // if set, used to compress logs for transmission
```

Comment:
I find this API a little confusing, at least without additional documentation that NewZstdEncoder and CompressTransport generally shouldn't be used together.

Is the new field really necessary?  Any reason not to just set SingleSegment in the two places where we are setting NewZstdEncoder?

Reply:
I'd rather move in the direction of hiding any details of zstd from the public API of `logtail`. It's an implementation detail and its exposure makes it harder to change the underlying protocol in the future.

It turns out that the only usages of `NewZstdEncoder` are in the `tsnet` and `logpolicy` package. I'm thinking of just deleting it so there's no ambiguity how the two options compose together.

Reply:
+1 to removing `NewZstdEncoder`